software-mansion/starknet-jvm
# RPC 0.6.0 cleanup

## Describe your changes


Refactoring and cleanup post #358. Includes missing transaction v3 functionalities, more streamlined interface, bug fixes and tests covering Starknet 0.13 features.


- `Account`, `StandardAccount`
  - all methods that have v1, v2 and v3 variants now have `V1`, `V2`, `V3` postfixes respectively
  - `estimateFeeV(1|3)` no longer accepts `Set<SimulationFlagForEstimateFee>`, instead takes `skipValidate: Bool`
  - Add missing `estimateFeeV3(Call)`
  - #396 
-  `Deployer`, `StandardDeployer`: Add `deployContractV3` methods ; rename `deployContract` ->`deployContractV1` 
- Rename `ExecutionParamsV3` -> `InvokeParamsV3`
- Restrict usage of v3 fields
  - `TransactionPayload`: make primary constructor private; add custom constructor with hardcoded values
  - `TransactionFactory`: don't accept v3 transaction params 
- Add `DeprecatedTransaction` and `TransactionV3` interfaces 
- #393  
- Fix declare v3 payload for fee estimate
- `StarknetChainId`: Rename `SEPOLIA_TESTNET`->`SEPOLIA`, renamed `SEPOLIA_INTEGRATOIN`->`INTEGATION_SEPOLIA` (should we remove enum altogether?)
- Add missing RPC 0.6.0 tests
  - `TransactionHashCalculator`: test v3s; add missing declare v1/v2 tests
  - Rename `FeltTests` -> `NumAsHexBaseTests `; test all numbers; nest Felt-specific tests; add Uint256 tets
  - `StandardAccountTest`
    - Nest test cases: add `DeclareEstimateTest`, `DeclareTests`, `DeployAccountEstimateTest`, `DeployAccountTest`, `DeployAccountEstimateTest`
    - Add missing v3 tests (`simulateTransactions`, `estimateFee`)
    - Add estimate fee with `SKIP_VALIDATE` test
  - `StandardDeployerTest`: test deploy v3
    - Remove `contract_with_constructor.cairo` (no longer used in tests)

## Linked issues



Closes #402
Closes #372 
Closes #393 
Closes #395 
Closes #396 
Closes #374 

## Breaking changes

- [x] This issue contains breaking changes


- `Account`, `StandardAccount`: all methods that have v1, v2 and v3 variants now have `V1`, `V2`, `V3` postfixes respectively
- `Deployer`, `StandardDeployer`: `deployContract` split into `deployContractV1` and `deployContractV3` 
- `estimateFeeV(1|3)` no longer accepts `Set<SimulationFlagForEstimateFee>`, instead takes `skipValidate: Bool`
- Removed transaction v3 parameters (`tip`, `paymasterData`, `accountDeploymentData`, `nonceDataAvailabilityMode`, `feeDataAvailabilityMode`) from `TransactionPayload` constructors
- Removed transaction v3 parameters (`tip`, `paymasterData`, `accountDeploymentData`, `nonceDataAvailabilityMode`, `feeDataAvailabilityMode`) from `TransactionFactory` methods
- Renamed `ExecutionParamsV3` -> `InvokeParamsV3`
- `StarknetChainId`: Renamed `SEPOLIA_TESTNET`->`SEPOLIA`, renamed `SEPOLIA_INTEGRATOIN`->`INTEGATION_SEPOLIA`,
